interpreter/syntax/types.ml

```diff
@@ -298,9 +316,13 @@ let unroll_def_type (dt : def_type) : sub_type =
   Lib.List32.nth sts i
 
 let expand_def_type (dt : def_type) : str_type =
```

Comment:
I would expect that this has to return desc_type now, since the clauses affect a type's structure and become relevant in some operations. With the flat type representation suggested above, they can just be pattern-matched away where not needed.

(FWIW, I should rename str_type to comp_type here, to sync with the spec.)